crates/client/rpc/src/lib.rs

```diff
@@ -517,4 +519,80 @@ where
             })?,
         })
     }
+
+    /// Gets a transaction details from it's hash.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_hash` - Hex string of the transaction hash.
+    ///
+    /// # Returns
+    ///
+    /// * `transaction` - transaction corresponding to the given hash, if any.
+    fn get_transaction_by_hash(&self, transaction_hash: FieldElement) -> RpcResult<RpcTransaction> {
+
+        let transaction_hash = H256::from_slice(&transaction_hash.to_bytes_be()[..32]);
+
+        let substrate_block_hash = self.backend
+            .mapping()
+            .read_transaction_block_hash(transaction_hash)
+            .map_err(|e| {
+                error!("Failed to get transaction's substrate block hash from mapping_db: {e}");
+                StarknetRpcApiError::TxnHashNotFound
+            })?;
+
+        if substrate_block_hash.is_none() {
+            return Err(StarknetRpcApiError::TxnHashNotFound.into());
+        }
+
+        let substrate_block_hash = substrate_block_hash.unwrap();
+
+        println!("-----> {:?}", substrate_block_hash);
+
+        let block = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), substrate_block_hash)
+            .current_block(substrate_block_hash)
+            .unwrap_or_default();
+
+        println!("-----> {:?}", block);
+
+        match block.transactions() {
+            BlockTransactions::Full(transactions) => {
+                for (txn, receipt) in transactions {
+                    if txn.hash != transaction_hash {
+                        continue;
+                    }
+
+                    let h = FieldElement::from_byte_slice_be(&txn.hash.as_bytes()[..32]).map_err(|e| {
+                        error!("{e}");
+                        StarknetRpcApiError::TxnHashNotFound                        
+                    })?;
+
+                    println!("------(TXN)----> {:?}", txn);
+
+                    let transaction = InvokeTransactionV1 {
+                        transaction_hash: h,
+                        max_fee: FieldElement::ZERO,
+                        signature: vec![FieldElement::ZERO],
+                        nonce: FieldElement::ZERO,
+                        sender_address: FieldElement::ZERO,
+                        calldata: vec![FieldElement::ZERO],
+                    };
+
+                    // Map the correct transaction base on the transaction type.
```

Comment:
The mapping here is really heavy. I am looking for alternative, if any. And add the TxType rework if I can as suggested by @EvolveArt.